Ruins/src/main/java/atomicstryker/ruins/common/RuinTemplateRule.java

```diff
@@ -606,7 +606,7 @@ else if (dataString.startsWith("HardChest"))
         else if (dataString.startsWith("ChestGenHook:"))
         {
             String[] s = dataString.split(":");
-            int targetCount = s.length > 2 ? Integer.valueOf(s[s.length-2].split("-")[0]) : 0;
+            int targetCount = s.length > 2 ? Integer.valueOf(s[s.length-1].split("-")[0]) : 0;
             addChestGenChest(world, random, x, y, z, s[1], targetCount, blocknum, rotate);

```

Comment:
You're only passing the namespace part of the loot table location, s[1], to addChestGenChest(). So, for example, a location of "neuroflow:chests/chestloot1" will be interpreted as "minecraft:neuroflow"...not what you want.

Maybe it'd be simpler to leave the existing code as-is and just massage the s array a bit, glomming the parts of the loot table location together, before using it? Something like:
```
String[] s = dataString.split(":");
if (s.length > 3)
{
    s[1] = String.join(":", s[1], s[2]);
    s[2] = s[3];
}
```

Reply:
Right,right, that will work.